Added Section "Plugins and High Availability" (#2989)
mattermost/docs: source/deployment/cluster.rst

:ref:`Run all job servers <command-line-tools-mattermost-jobserver>` with ``--noschedule flag``, then set ``JobSettings.RunScheduler`` to ``true`` in config.json for all app servers in the cluster. The cluster leader will then be responsible for scheduling recurring jobs.

Plugins and High Availability
^^^^^^^^^^^^^^^^

As of Mattermost 5.14, when you install or upgrade a plugin, it is propagated across the servers in the cluster automatically. 

> Note a slight behavior change in v5.15:

Reviewer: ```suggestion
Note a slight behaviour change in 5.15:
```

Author: @amyblais Interesting, do we use Canadian English or American? for future reference.

> When you re-install a plugin in 5.14, it retained the last `Enabled` or `Disabled` state as it was before. As of 5.15, when you re-install a plugin it's initial state will now be DISABLED.

Reviewer: ```suggestion
When you re-install a plugin in 5.14, it retained the last "Enabled" or "Disabled" state as it was before. As of 5.15, a plugin's initial state will now be DISABLED when you re-install it.
```

> When you re-install a plugin in 5.14, it retained the last "Enabled" or "Disabled" state as it was before. As of 5.15, a plugin's initial state will now be DISABLED when you re-install it.

Reviewer: ```suggestion
When you re-install a plugin in 5.14, it retained the last "Enabled" or "Disabled" state as it was before. As of 5.15, a plugin's initial state will now be disabled when you re-install it.
```

> When you install a plugin in 5.14 -> enable -> uninstall -> install the same plugin again, the plugin's initial state remained  enabled. As of v5.15, when you install a plugin -> enable -> uninstall -> install the same plugin again, the plugin's initial state will now be DISABLED.

Reviewer: ```suggestion
When you install a plugin in 5.14 via **enable > uninstall > install the same plugin again**, the plugin's initial state remained  enabled. As of 5.15, when you install a plugin via **enable > uninstall > install the same plugin again**, the plugin's initial state will now be disabled.
```

Reviewer: This might need more details on where the "enable > uninstall > install the same plugin again" settings are.

Reviewer 2: Does this seem awfully detailed? Should we rather say something like: `Plugins retained their previous state on reinstall. As of 5.15, plugins are always disabled on install.` Or something like that -- not a word wizard unfortunately :P

Reviewer 2: Perhaps @justinegeffen can help us on her first day xD 👋

Author: I re-worded it to simplify.

Author: @justinegeffen Please, let me know if you have any suggestions.
